sw/common/spi.c

```diff
@@ -0,0 +1,26 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+#include "spi.h"
+#include "dev_access.h"
+
+typedef struct spi_handle{
+    spi_reg_t *reg;
+    uint32_t speed;
+}spi_handle_t;
+
+void spi_init(spi_t *spi, spi_reg_t *spi_reg, uint32_t speed){
+  spi->reg = spi_reg;
+  spi->speed = speed;
+}
+
+void spi_send_byte_blocking(spi_t *spi, char c) {
+  spi->reg->tx_reg = c;
+  while(spi->reg->status_reg & SPI_STATUS_TX_FULL);
```

Comment:
Better off to have the `while` first? Then we're guaranteed space in the transmit FIFO when we write to it